woocommerce/woocommerce
# Fix post-merge CI / update ci-jobs util to make base-ref optional

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



This PR updates the `ci-jobs` utility such that `baseRef` is no longer a required argument and is now an optional argument. It also removes the `force` option. When no `baseRef` is supplied, `ci-jobs` acts as it did previously when provided the `force` option, returning all jobs for processing. When `baseRef` is supplied, HEAD is compared with baseRef and the changes determine which jobs are returned.



Closes #43679 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a branch based on this branch. Make some changes to a file in one of the projects that's included in the CI config within that project's json. For example, add a comment to `packages/js/onboarding/src/utils/countries/index.ts`.
2. Commit those changes to the branch.
3. Switch back to this branch. Run `pnpm utils ci-jobs`. This should return all jobs.
4. Run `pnpm utils ci-jobs --base-ref heads/YOURBRANCHNAME`, replacing `YOURBRANCHNAME` with your branch name. Only the jobs relating to your project should be returned (keeping in mind that changes in dependencies trigger the jobs of their parent). If you updated `packages/js/onboarding/src/utils/countries/index.ts`, this should return jobs from `@woocommerce/onboarding` and `@woocommerce/admin-library` (the latter because onboarding is a dependency of admin-library).
5. Verify that CI passes on this PR and the matrix created by `ci.yml` only runs jobs from monorepo utils. 



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
